src/app/organizations/state/collections.service.ts

```diff
@@ -157,4 +161,47 @@ export class CollectionsService {
         }
       );
   }
+
+ /**
+  * Deletes the specified collection
+  * @param organizationId
+  * @param collectionId
+  * @param organizationName
+  * @param collectionName
+  */
+  deleteCollection(organizationId: number, collectionId: number, organizationName: string, collectionName: string) {
+    // Implement the store setLoading/Error and alertService calls per the recommendations here:
+    // https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#notifications-user-feedback
+    this.collectionsStore.setLoading(true);
+    this.collectionsStore.setError(false);
+    this.alertService.start('Removing collection ' + collectionName);
+    this.openApiOrganizationsService
+      .deleteCollection(organizationId, collectionId)
+      .subscribe(
+        () => {
+          this.collectionsStore.setLoading(false);
+          this.collectionsStore.setError(false);
+          this.alertService.detailedSuccess();
+          this.matDialog.closeAll();
+          // Navigate to the organization page.
+          // In the following if-else statement, both code paths trigger http requests.
+          // Do one or the other to avoid duplicate and possibly-overlapping requests and store updates.
+          if (this.router.url.endsWith('/organizations/' + organizationName)) {
+            // We're already on the organization page, so router.navigate will not trigger noOnInit/etc.
+            // Update the state manually.
```

Comment:
I changed the comments to describe the two reasonable use cases, but since it's not unlikely that both might see use in the future, I left it at that, to avoid introducing a comment that might become stale.